### PR TITLE
Fixed Termination Manager logging to report aggregated percentage of environments done due to each term. 

### PR DESCRIPTION
**Pull request by ooctipus**
# Description

Currently Termination Manager write current step's done count for each term if reset is detected. This leads to two problem. 
1. User sees different counts just by varying num_envs
2. the count can be over-count or under-count depending on when reset happens, as pointed out in #2977 (Thanks, @Kyu3224)

The cause of the bug is because we are reporting current step status into a buffer that suppose to record episodic done. So instead of write the entire buffer base on current value, we ask the update to respect the non-reseting environment's old value, and instead of reporting count, we report percentage of environment that was done due to the particular term.

Test on Isaac-Velocity-Rough-Anymal-C-v0

Before fix:
Screenshot from 2025-08-06 22-16-20
Red: num_envs = 4096, Orange: num_envs = 1024

After fix:

Screenshot from 2025-08-06 22-16-12
Red: num_envs = 4096, Orange: num_envs = 1024

Note that curve of the same color ran on same seed, and curves matched exactly, the only difference is the data gets reported in termination.  The percentage version is a lot more clear in conveying how agent currently fails, and how much percentage of agent fails, and shows that increasing num_envs to 4096 helps improve agent avoiding termination by `base_contact` much quicker than num_envs=1024. Such message is a bit hard to tell in first image.

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab/isaaclab/managers/termination_manager.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def get_term(self, name: str) -> torch.Tensor:
         Returns:
             The corresponding termination term value. Shape is (num_envs,).
         """
-        return self._term_dones[name]
+        return self._term_dones[name, self._term_names.index(name)]
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/managers/termination_manager.py]: This indexing here is not great. We use the value from here to give termination rewards in different environments. Doing the `.index` every time for reward computation may lead to slow downs (as it searches over the list in O(n) fashion).

File: source/isaaclab/isaaclab/managers/termination_manager.py [hunk 2]
```diff
@@ -61,9 +61,7 @@ def __init__(self, cfg: object, env: ManagerBasedRLEnv):
         # call the base class constructor (this will parse the terms config)
         super().__init__(cfg, env)
         # prepare extra info to store individual termination term information
-        self._term_dones = dict()
-        for term_name in self._term_names:
-            self._term_dones[term_name] = torch.zeros(self.num_envs, device=self.device, dtype=torch.bool)
+        self._term_dones = torch.zeros((self.num_envs, len(self._term_names)), device=self.device, dtype=torch.bool)
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/managers/termination_manager.py:65]: Is there a reason to change this as a dict of tensors to a single tensor?

File: source/isaaclab/isaaclab/managers/termination_manager.py [hunk 3]
```diff
@@ -139,9 +137,10 @@ def reset(self, env_ids: Sequence[int] | None = None) -> dict[str, torch.Tensor]
             env_ids = slice(None)
         # add to episode dict
         extras = {}
-        for key in self._term_dones.keys():
+        last_episode_done_stats = self._term_dones.float().mean(dim=0)
+        for i, key in enumerate(self._term_names):
             # store information
-            extras["Episode_Termination/" + key] = torch.count_nonzero(self._term_dones[key][env_ids]).item()
+            extras["Episode_Termination/" + key] = last_episode_done_stats[i].item()
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/managers/termination_manager.py:144]: If you want the ratio, isn't that simply?

```python
self._term_dones[key][env_ids].sum() / len(env_ids)
```

Comment (ooctipus) [reply source/isaaclab/isaaclab/managers/termination_manager.py:144]: Thanks for reveiwing!
I guess, doing with env_ids will be viewing ratio of resetting environments. I thought maybe report stats of all environment can be a bit more nicer as user can verify from the graph that all terms sum up to 1. Of course you can do `self._term_dones[key].sum() / self.env.num_envs` as well

But it seems like if this approach is what I after, do it in one tensor operation seems quite nice, both speed wise and memory utility wise.

File: source/isaaclab/isaaclab/managers/termination_manager.py (hunk 1 above)
Comment (ooctipus) [reply source/isaaclab/isaaclab/managers/termination_manager.py]: Ahhh if you like the single tensor approach I can also store one more name -> idx dict, and make change here O(1), I wasn't aware of termination rewards, (my bad!!) and thought this function may be used infrequently. I should do a global search next time rather than assume!

File: source/isaaclab/isaaclab/managers/termination_manager.py (hunk 2 above)
Comment (ooctipus) [reply source/isaaclab/isaaclab/managers/termination_manager.py:65]: I thought it `last_episode_done_stats = self._term_dones.float().mean(dim=0)` this operation is very nice and optimized, thats why I did it, but if you think dict is more clear I can revert back : ))

**Comment by ooctipus** [comment]
@Mayankm96 
I did my best to optimize the operations in compute and reset,
index all changed from O(n) to O(1) -> this part not captured in benchmark below

benchmarked task: velocity rough anymal c, 4096 envs, 1000 steps

before change :
one step total: 79.038
```
| termination.compute     |       0.229 ms|
| termination.reset       |       0.007 ms|
```
after change:

one step total: 80.187 ms
```
| termination.compute     |       0.274 ms|  slower due to one extra operation
| termination.reset       |       0.004 ms|  twice as fast
```

the cost for this PR is about 0.04 ms / 80, 0.05%, mostly due to compute needs modify other terms's done to correctly update the done buffer. I think this is reasonable
